### DISCUSSION TITLE
Alpine JS - $wire is not defined

### DISCUSSION THREAD

**Discussion by jeroenvanrensen** [livewire/livewire#2415] [1. Help] [answered]
Hi everyone,

I'm using the TALL stack to create a website.

I have this code in my Livewire component:

```html
<script>
    $wire.on('createdLink', () => {
        alert(1);
    });
</script>
```

And this is my PHP code:

```php
$this->emit('createdLink');
```

Here I'm loading in the scripts (in the `head` section):

```html
<livewire:scripts />
<script src="https://cdn.jsdelivr.net/gh/alpinejs/alpine@v2.8.0/dist/alpine.min.js" defer></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/turbolinks/5.2.0/turbolinks.js"></script>
<script src="https://cdn.jsdelivr.net/gh/livewire/turbolinks@v0.1.x/dist/livewire-turbolinks.js" data-turbolinks-eval="false"></script>
```

But when I load the page I get this error in my console:

```
Uncaught ReferenceError: $wire is not defined
```

Does anyone know what's happening here? I searched about it but couldn't find anything useful.

Thank you! Jeroen

**Comment by ju5t** [comment, marked as answer]
`$wire` only works within an AlpineJS component. See: https://laravel-livewire.com/docs/2.x/alpine-js

**Reply by jeroenvanrensen** [reply]
Okay thanks!

**Comment by alitokmakci** [comment]
You can listen event in javascript with Livewire not with $wire.

So you should do:

```
<script>
    Livewire.on('createdLink', () => {
        alert(1);
    });
</script>
```

Docs: [Listening For Events In JavaScript](https://laravel-livewire.com/docs/2.x/events#in-js)

**Reply by jeroenvanrensen** [reply]
Yeah but I want to do it **inside** an Alpine.JS component.

**Reply by alitokmakci** [reply]
Oh my bad I thought you were trying to access in vanilla JS! lol

**Comment by TwilightDuck** [comment]
I know this is an old topic but for anyone who stumbles on this.
Try calling `window.Livewire.start()` before `Alpine.start()`

Like so:

```
if (window.Livewire) {
    window.Livewire.start();
}

Alpine.start();
```

**Comment by GregOstry** [comment]
I've recently installed Laravel and am encountering an issue. My setup is as follows:

Laravel Version: 10.40.0
PHP Version: 8.3
AlpineJS: 3.4.2
Here's a simple example I'm working with:

```
<div>
    <div x-data>
        <h1 x-text="$wire.count"></h1>

        <x-secondary-button x-on:click="$wire.increment()">+</x-secondary-button>
    </div>
</div>
```

However, I'm facing an error: `Alpine Expression Error: $wire is not defined.`

Also, when I try to insert `window.Livewire.start()` before `Alpine.start()`, I encounter another error: `Uncaught TypeError: Cannot redefine property: $persist.`

I'm unsure about what I might be missing in my configuration or code. Is there a specific step or setting that I need to ensure is correctly implemented for Livewire and AlpineJS to work together seamlessly in Laravel? Any advice or insights would be greatly appreciated.

**Reply by joshhanley** [reply]
@GregOstry make sure you are not running Alpine from npm and instead use it from the bundled Livewire version in your `app.js`. See here for details https://livewire.laravel.com/docs/installation#manually-bundling-livewire-and-alpine

**Reply by GregOstry** [reply]
Firstly, I would like to thank you for the prompt response @joshhanley. 

I believe I've managed to identify the problem. 
Initially, I installed Laravel Breeze (without Livewire), which added Alpine.js to the app.js file, as seen here: [https://github.com/laravel/breeze/blob/1.x/stubs/default/resources/js/app.js](https://github.com/laravel/breeze/blob/1.x/stubs/default/resources/js/app.js).

Subsequently, I decided to also install Laravel Livewire, resulting in Alpine.js being loaded through app.js before livewire.js.

Thank you for the tip!

**Reply by joshhanley** [reply]
@GregOstry no worries! 🙂 yep, Breeze uses Alpine so it pulls it in. But yeah if you add Livewire later, you need to remove Alpine and instead use the version bundled with Livewire
